attoparsec-iso8601/src/Data/Attoparsec/Time.hs

```diff
@@ -72,6 +82,17 @@ quarter = do
       <|> Q3 <$ char '3'
       <|> Q4 <$ char '4'
 
+-- | Parse a year @YYYY@, with at least 4 digits. Does not include any sign.
+year :: Parser Year
+year = do
+  ds <- takeWhile1 isDigit
+  if T.length ds < 4 then
+    fail "expected year with at least 4 digits"
+  else
+    pure (T.foldl' step 0 ds)
```

Comment:
This is bad. See https://github.com/haskell/aeson/issues/673 and its fix.

Reply:
uhoh. Thanks for catching this.

Reply:
BTW `decimal` is also used here:

https://github.com/haskell/aeson/blob/65fca856304c0b323d538cc783493d8d7a145114/src/Data/Aeson/Parser/Internal.hs#L498

that  should be removed too right?

Reply:
`scientific :: Integer -> Int -> Scientific` so that `decimal` is parsing `Int` and is fine.

Reply:
It won't run in quadratic time, so no DoS, but it will still overflow with a wrong result, which is undesirable.